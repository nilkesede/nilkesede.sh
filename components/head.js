import React, {Component} from 'react';
import NextHead from 'next/head';
import {string} from 'prop-types';

export default class Head extends Component {
  static defaultProps = {
    title: '',
    description: ''
  };

  static propTypes = {
    title: string,
    description: string
  };

  render() {
    const {title, description} = this.props;

    return (
      <NextHead>
        <meta charSet="UTF-8"/>
        <title>{title}</title>

        <meta name="description" content={description}/>
        <meta name="viewport" content="width=device-width, initial-scale=1"/>

        <link rel="icon" href="/static/favicon.ico"/>
      </NextHead>
    );
  }
}
