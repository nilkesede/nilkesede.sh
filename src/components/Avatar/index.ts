import styled from 'styled-components'

const Avatar = styled.img`
  width: 200px;
  border-radius: 50%;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.5);
`

export default Avatar
