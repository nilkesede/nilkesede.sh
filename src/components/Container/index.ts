import styled from 'styled-components'

const Container = styled.div`
  max-width: 1140px;
  display: flex;
  align-items: flex-start;
  justify-content: center;
  flex-direction: column;
  padding: 50px 30px 30px;
  flex-grow: 1;
`

export default Container
