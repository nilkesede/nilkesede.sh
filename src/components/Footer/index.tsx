import {Wrapper, Sign, Social, Anchor, Icon} from './styles'

const year = new Date().getFullYear()

const Footer = () => (
  <Wrapper>
    <Sign>Nil KÃ©sede &copy; {year}</Sign>
    <Social>
      <Anchor href="https://linkedin.com/in/nilkesede" title="linkedin">
        <Icon>&#xf0e1;</Icon>
      </Anchor>
      <Anchor href="https://github.com/nilkesede" title="github">
        <Icon>&#xf09b;</Icon>
      </Anchor>
      <Anchor href="https://twitter.com/nilkesede" title="twitter">
        <Icon>&#xf099;</Icon>
      </Anchor>
    </Social>
  </Wrapper>
)

export default Footer
