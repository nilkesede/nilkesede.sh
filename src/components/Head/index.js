import NextHead from 'next/head'
import {string} from 'prop-types'

const Head = ({title, description}) => (
  <NextHead>
    <meta charSet="UTF-8"/>
    <title>{title}</title>

    <meta name="description" content={description}/>
    <meta name="viewport" content="width=device-width, initial-scale=1"/>
    <meta name="theme-color" content="#222"/>

    <link rel="manifest" href="/manifest.json"/>
    <link rel="icon" href="/favicon.ico"/>
  </NextHead>
)

Head.propTypes = {
  title: string.isRequired,
  description: string
}

Head.defaultProps = {
  description: ''
}

export default Head
