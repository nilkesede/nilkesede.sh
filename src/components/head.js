import React, {Component} from 'react';
import NextHead from 'next/head';
import {string} from 'prop-types';
import '../styles/main.scss';

export default class Head extends Component {
  static propTypes = {
    title: string.isRequired,
    description: string
  };

  static defaultProps = {
    description: ''
  };

  render() {
    const {title, description} = this.props;

    return (
      <NextHead>
        <meta charSet="UTF-8"/>
        <title>{title}</title>

        <meta name="description" content={description}/>
        <meta name="viewport" content="width=device-width, initial-scale=1"/>
        <meta name="theme-color" content="#222"/>

        <link rel="manifest" href="/manifest.json"/>
        <link rel="icon" href="/favicon.ico"/>
      </NextHead>
    );
  }
}
