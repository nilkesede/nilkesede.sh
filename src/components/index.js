import Head from './Head'
import TagCloud from './TagCloud'

export {
  Head,
  TagCloud
}

export default {
  Head,
  TagCloud
}
