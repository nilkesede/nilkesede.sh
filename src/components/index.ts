import Avatar from './Avatar'
import Container from './Container'
import Footer from './Footer'
import Head from './Head'
import Subtitle from './Subtitle'
import Title from './Title'
import Wrapper from './Wrapper'

export {
  Avatar,
  Container,
  Footer,
  Head,
  Subtitle,
  Title,
  Wrapper
}
