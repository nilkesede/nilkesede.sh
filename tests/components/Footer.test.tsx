import {Footer} from '../../src/components'

import {render} from '../utils'

it('renders correctly', () => {
  const {container} = render(<Footer />)
  expect(container).toMatchSnapshot()
})
