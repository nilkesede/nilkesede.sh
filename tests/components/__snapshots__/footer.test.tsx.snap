// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c3 {
  width: 32px;
  height: 32px;
  font-size: 25px;
  padding: 0 5px;
  margin: 0 2.5px;
  display: inline-block;
}

.c4 {
  font-family: 'fontello',sans-serif;
  font-style: normal;
  font-weight: normal;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-top: 1px solid rgba(255,255,255,0.1);
  background-clip: padding-box;
  margin-top: 50px;
  padding: 20px 30px;
  overflow: hidden;
}

.c1 {
  font-size: 12px;
}

.c2 {
  margin-right: -7.5px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Nil Késede © 
      2020
    </div>
    <div
      class="c2"
    >
      <a
        class="c3"
        href="https://linkedin.com/in/nilkesede"
        rel="noopener noreferrer"
        target="_blank"
        title="linkedin"
      >
        <i
          class="c4"
        >
          
        </i>
      </a>
      <a
        class="c3"
        href="https://github.com/nilkesede"
        rel="noopener noreferrer"
        target="_blank"
        title="github"
      >
        <i
          class="c4"
        >
          
        </i>
      </a>
    </div>
  </div>
</div>
`;
